WcaOnRails/app/models/user.rb

```diff
@@ -841,8 +841,11 @@ def self.search(query, params: {})
       end
     end
 
+    query_options = "name LIKE :part OR wca_id LIKE :part"
+    query_options << " OR email LIKE :part" if params[:email]
```

Comment:
Our strings are [frozen](https://freelancing-gods.com/2017/07/27/an-introduction-to-frozen-string-literals.html) by default, so this won't work :s

You can try something like:
```ruby
query_options = "name LIKE :part OR wca_id LIKE :part #{"OR email LIKE :part" if params[:email]}"
```

Is there any other place where we use this `User.search` method and where we don't want to search by email?
(the follow up question would be: "can't we add the 'OR' statement regardless of a parameter?" :p)

Reply:
@viroulep I've corrected the error, please review the new changes

Reply:
Oh I see that's tricky because earlier in the method we choose to search on `Person` or on `User`, sorry I didn't notice that.
While it works very well with your implementation, I think we should make it clearer to the reader what's going on here.
I would suggest to define a "search_by_email" variable to false, that we update to `params[:email]` in the `unless` block.
Basically the code would look like:

```ruby
    users = Person.includes(:user).current
    # We can't search by email on the 'Person' table
    search_by_email = false
    unless ActiveRecord::Type::Boolean.new.cast(params[:persons_table])
      users = User.confirmed_email.not_dummy_account
      search_by_email = ActiveRecord::Type::Boolean.new.cast(params[:email])
      #...   
    end

    query.split.each do |part|
      users = users.where("name LIKE :part OR wca_id LIKE :part #{"OR email LIKE :part" if search_by_email}", part: "%#{part}%")
    end
    #...
```


(again sorry for misleading you during my initial review and having you make some additional changes :s)

Reply:
Good idea! I've added the changes